### DISCUSSION TITLE
Switch to Pulseaudio

### DISCUSSION THREAD

**Discussion by s-martin** [MiczFlor/RPi-Jukebox-RFID#1208] [Ideas]
Raspberry Pi’s December 2020 update switched to PulseAudio (with still underlaying ALSA), see https://www.raspberrypi.org/blog/new-raspberry-pi-os-release-december-2020/

I want to start a discussion, if it’s helpful or necessary to switch to PulseAudio. I don’t know much about Linux audio (except that it’s a mess).

Opinions, ideas?

**Comment by mattelacchiato** [comment]
I've just tried to install pulseaudio. It heavily depends on x11/dbus. This can be faked via Xvfb, but at least on a Raspberry Pi Zero, this maybe would be too much overhead?
My starting point was this: https://unix.stackexchange.com/questions/105964/launch-a-fake-minimal-x-session-for-pulseaudio-dbus

RPi-Jukebox-RFID currently does not work at all when using a latest "lite" image of Raspbian / Raspberry Pi OS. Neither alsa, nor pulseaudio is installed by default.

**Reply by s-martin** [reply]
Thanks for investigating this. 
Currently I see no benefit using PulseAudio, so I wouldn't want to add a heavy dependency like x11.


PR #1255 should make sure ALSA is always installed.

**Reply by Schneelocke** [reply]
I also tried to use pulseaudio for bluetooth playback instead bluealsa. Ended up with double the load average on my Pi Zero. 
So switched back to Alsa. :)

**Reply by mattelacchiato** [reply]
I think, we could close this ticket then?

**Reply by s-martin** [reply]
That's a discussion, it's not supposed to be closed.

**Reply by charlesvestal** [reply]
One use case I'm trying to implement is creating a box that outputs to airplay speakers (i.e. homepod). It seems I could do this with something like outputting to a [loopback device and streaming that directly ](https://blog.vivekpanyam.com/streaming-system-audio-to-an-airplay-receiver/), but I'd imagine [pulseaudio-raop2](https://github.com/hfujita/pulseaudio-raop2) to be a cleaner solution.

**Reply by pabera** [reply]
We are currently developing a new version under the future3 branches. The latest release includes pulseaudio as the primary sound manager. And indeed it's possible to connect to Bluetooth speakers as well which is described in our documentation. You might be able to connect it to your Homepod as well, with some tweaks :-)

**Comment by pabera** [comment]
Version 3.x is running on PulseAudio, supporting various sound backends like USB Sound Cards, Hifi Dacs or Bluetooth speakers.
